prepare-release.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/sh
+
+set -eux
+
+version=$1
+
+if ! echo "$version" | grep -q "^[[:digit:]]\+\.[[:digit:]]\+\.[[:digit:]]\+$"; then
+  echo "argv[1] needs to be a version, in x.y.z format."
+  exit 1
+fi
+
+git fetch
+git checkout origin/main
+git branch -D "release-v$version" || true
+git checkout -b "release-v$version"
+
+sed -i '/^version = ".*"$/s/^.*/version = "'"$version"'"/' Cargo.toml
+git add Cargo.toml
+
+for fname in $(find ./tests/fixtures -name '*.json'); do
+  cat "$fname" \
+    | jq '.version = $version | .diagnostic_data.version = $version' --arg version "$version" \
+    > "$fname.next"
+  mv "$fname.next" "$fname"
+  git add "$fname"
+done
+
+git commit -m "Update Cargo.toml and fixtures to v$version"
+
+cargo update --aggressive
+git add Cargo.lock
+git commit -m "Update Cargo.lock prior to v$version"
```

Comment:
These should be in the reverse order. Just update cargo.lock, _then_ update cargo.toml and run `cargo fetch` to update the lockfile with the new crate version afterwards.

Reply:
Ok, sure. m'onit.

Reply:
In fact, let's do the version bump last and do the flake.lock and cargo.lock updates first please.